File: src/main/java/org/alham/alhamfirst/repository/stat/UserStatRepository.java
package org.alham.alhamfirst.repository.stat;

import org.alham.alhamfirst.document.stat.UserStatDocument;
import org.springframework.data.mongodb.repository.MongoRepository;

public interface UserStatRepository extends MongoRepository<UserStatDocument, String> {
    UserStatDocument findByUserId(long userId);
}

File: src/main/java/org/alham/alhamfirst/repository/stat/UserStatRepository.kt
package org.alham.alhamfirst.repository.stat;

import org.alham.alhamfirst.domain.document.stat.UserStatDocument;

interface UserStatRepository{
    fun findByUserId(userId: Long): UserStatDocument
    fun updateUserStat(userId: Long, statData: Map<String,Double>, completed: Boolean): UserStatDocument
    fun createUserStatDocument(userId: Long): UserStatDocument
}
